### PR TITLE
Add shortcut Shift + F3 to search pervious in the built-in docs

### LINKED ISSUE #31147: Shift + F3 doesn't work in the built-in docs
**Godot version:**
3.1.1

**Issue description:**
When you search something in the documentation, you can use F3, but Shift + F3 (search previous) doesn't work.

**Steps to reproduce:**
1. Open some doc in the editor
2. Search something
3. Press F3
4. Press Shift + F3
5. Press Shift + F3 again, just to be sure

### PR DESCRIPTION
**Pull request by MarqusJonsson**
When using the built-in docs, Godot would not support the shortcut "Shift + F3"
to search for the previous occurrence of the search entry text, thus causing an
inconsistent behaviour when using shortcuts in the "ScriptEditor" compared to
using them in the "ScriptTextEditor".

The previous parameter of the function "EditorHelp::_search()" in the class
"editor_help" seems to be unused, thus replaced with a bool representing to
search for previous search entry text or not. By adding the shortcut to
Godot's "ScriptEditor", this commit now improves Godot's consistensy when
using shortcuts.

Fixes #31147.

Co-Authored-By: Oscar Ferm <oscfer-6@student.ltu.se>

### CODE REVIEW

**Comment by MarqusJonsson** [comment]
While testing this we also found a bug related to the search previous function within the built-in docs. please see #32687.

File: editor/plugins/script_editor_plugin.cpp [hunk 1]
```diff
@@ -1325,6 +1325,9 @@ void ScriptEditor::_menu_option(int p_option) {
 				case HELP_SEARCH_FIND_NEXT: {
 					help->search_again();
 				} break;
+				case HELP_SEARCH_FIND_PREVIOUS: { // ADDED
```
Comment (akien-mga) [inline editor/plugins/script_editor_plugin.cpp]: You should not add `// ADDED` comments, that what we use Git for :)

File: editor/plugins/script_editor_plugin.cpp [hunk 2]
```diff
@@ -2826,7 +2829,8 @@ void ScriptEditor::_update_selected_editor_menu() {
 	if (eh) {
 
 		script_search_menu->get_popup()->add_shortcut(ED_SHORTCUT("script_editor/find", TTR("Find..."), KEY_MASK_CMD | KEY_F), HELP_SEARCH_FIND);
-		script_search_menu->get_popup()->add_shortcut(ED_SHORTCUT("script_editor/find_next", TTR("Find Next"), KEY_F3), HELP_SEARCH_FIND_NEXT);
+		script_search_menu->get_popup()->add_shortcut(ED_SHORTCUT("script_editor/find_next", TTR("Find Dext"), KEY_F3), HELP_SEARCH_FIND_NEXT);
```
Comment (akien-mga) [inline editor/plugins/script_editor_plugin.cpp]: Typo, should be "Find **N**ext"

File: editor/plugins/script_editor_plugin.cpp [hunk 3]
```diff
@@ -2826,7 +2829,8 @@ void ScriptEditor::_update_selected_editor_menu() {
...
+		script_search_menu->get_popup()->add_shortcut(ED_SHORTCUT("script_editor/find_previous", TTR("Find previous"), KEY_MASK_SHIFT | KEY_F3), HELP_SEARCH_FIND_PREVIOUS); // ADDED
```
Comment (akien-mga) [inline editor/plugins/script_editor_plugin.cpp]: Should be "Find **P**revious"

File: editor/plugins/script_text_editor.cpp [hunk 4]
```diff
@@ -1939,11 +1939,11 @@ void ScriptTextEditor::register_editor() {
 
 	ED_SHORTCUT("script_text_editor/find", TTR("Find..."), KEY_MASK_CMD | KEY_F);
 #ifdef OSX_ENABLED
-	ED_SHORTCUT("script_text_editor/find_next", TTR("Find Next"), KEY_MASK_CMD | KEY_G);
+	ED_SHORTCUT("script_text_editor/find_next", TTR("Find Oext"), KEY_MASK_CMD | KEY_G);
```
Comment (akien-mga) [inline editor/plugins/script_text_editor.cpp]: Er...

File: editor/plugins/script_text_editor.cpp [hunk 5]
```diff
@@ -1939,11 +1939,11 @@ void ScriptTextEditor::register_editor() {
...
 	ED_SHORTCUT("script_text_editor/find_previous", TTR("Find Previous"), KEY_MASK_CMD | KEY_MASK_SHIFT | KEY_G);
 	ED_SHORTCUT("script_text_editor/replace", TTR("Replace..."), KEY_MASK_ALT | KEY_MASK_CMD | KEY_F);
 #else
-	ED_SHORTCUT("script_text_editor/find_next", TTR("Find Next"), KEY_F3);
+	ED_SHORTCUT("script_text_editor/find_next", TTR("Find Texn"), KEY_F3);
```
Comment (akien-mga) [inline editor/plugins/script_text_editor.cpp]: Nope.

**Review by MarqusJonsson — COMMENTED** [review]
These mistakes were all addressed but did not get through in the first commit after commiting the wrong files. Thank you for noticing, will try to be be more careful in the future.

**Comment by akien-mga** [comment]
Changes look good, could you squash the commits together? See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**